include/pybind11/numpy.h

```diff
@@ -655,23 +655,18 @@ class array : public buffer {
     }
 
     /// Resize array to given shape
-    /// If refcheck is true, then resize will fail if > 1 references exists to this array
-    void resize(const std::vector<size_t> &new_shape, bool refcheck = false) {
+    /// If refcheck is true and more that one reference exist to this array
+    /// then resize will succeed only if it makes a reshape, i.e. original size doesn't change
+    void resize(const std::vector<size_t> &new_shape, bool refcheck = true) {
```

Comment:
I've merged #788, so you should be able to switch the input to a `ShapeContainer new_shape` now.  (Unfortunately I also caused some conflicts--both had tests added in the same place--but I added a commit to hopefully resolve them).